### PR TITLE
refactored notification mail address & now sending notifications for …

### PR DESCRIPTION
**Pull request by TheRittler**
…new added giftcodes

### CODE REVIEW

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java [hunk 1]
```diff
@@ -297,26 +319,44 @@ private void payWorker(WorkerRecord worker, List<GiftCodeRecord> giftCodes) thro
...
+            InputStreamReader reader;
+            switch (level) {
+                case ERROR:
+                    reader = new InputStreamReader(Main.class.getResourceAsStream("/moneytransfer/errorNotification.properties"));
+                    break;
+                case ISSUE:
+                    reader = new InputStreamReader(Main.class.getResourceAsStream("/moneytransfer/issueNotification.properties"));
+                    break;
+                case GIFTCODES_ADDED:
+                    reader  = new InputStreamReader(Main.class.getResourceAsStream("/moneytransfer/giftcodesAddedNotification.properties"));
```
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java]: ^ missing break

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java [hunk 2]
```diff
@@ -297,26 +319,44 @@ private void payWorker(WorkerRecord worker, List<GiftCodeRecord> giftCodes) thro
...
+            throw new MoneyTransferException("Error while loading properties for a notification.", e);
         }
+        String subject = properties.getProperty("subject");
+        String mail = Utils.loadFile(properties.getProperty("pathToMessage"));
 
-        mail = mail.append(message);
+        Map<String, String> map = new HashMap<>();
+        map.put("content", message);
+        mail = Template.apply(mail, map);
+        System.out.println(mail);
```
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java]: öhm, debugging stuff?

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java [hunk 3]
```diff
@@ -297,26 +319,44 @@ private void payWorker(WorkerRecord worker, List<GiftCodeRecord> giftCodes) thro
...
         try {
-            if (notificationText.length() != 0) {
-                LOGGER.trace("Started sending a notification about problems with submission of giftcodes.");
-                mailSender.sendMail(notificationMailAddress, subject, mail.toString());
-                LOGGER.trace("Completed sending a notification about problems with submission of giftcodes.");
+            if (mail.length() != 0) {
+                LOGGER.trace("Started sending a notification about " + properties.getProperty("loggerMessage") + ".");
+                mailSender.sendMail(notificationMailAddress, subject, mail);
+                LOGGER.trace("Completed sending a notification about " + properties.getProperty("loggerMessage") + ".");
```
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java]: Why would you log this ?

Comment (TheRittler) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java]: I guess its interesting to know, if an error occurrs with moneytransfer if the user got a notification mail or not or if it was tried to send a mail. I think we can never log to much in moneytransfer.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java (hunk 1 above)
Comment (TheRittler) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java]: added

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java (hunk 3 above)
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java]: The reason why I am asking is that adding propertie files just for something like this feels like a unusual solution. 
Why not atting something like that:

```
public enum ErrorMessage {
ERROR("Hard mistake happend", "path/to/message", 
NOTIFY("Notification", "blba/bla/bla"),

private final String name;
private final String path;

public ErrorMessage(String displayName, String path) {
...
}
}
```

For more details: http://stackoverflow.com/questions/2457076/can-i-add-a-function-to-enums-in-java

**Comment by marcelhollerbach** [comment]
Okay, I am okay with it, one more question, why have you moved the mailadress to the mail section, is this address used somewhere outside of moneytransfer?

**Comment by TheRittler** [comment]
simon is going to use it for notifications
so please merge fast, he needs the changes :wink:

**Comment by marcelhollerbach** [comment]
mhmm I have to say that I quite dont like that, for what is it used in notifications ?

**Comment by marcelhollerbach** [comment]
Since we are in a rush

**Comment by kelunik** [comment]
@marcelhollerbach That's fine and planned. @phimonk has to send emails to the admin. We still need another config to configure the sender, not just the receiver.

**Comment by marcelhollerbach** [comment]
You wanted to have a receiver ? Then we would need a fetcher/sender configuration not just a single emailadress. Also is it possible to tell the "plan" ?

**Pull request merged**